LibraBFT/Impl/Consensus/SafetyRules/SafetyRules.agda

```diff
@@ -96,8 +96,8 @@ verifyAndUpdatePreferredRoundM quorumCert safetyData = do
 verifyAuthorM : Maybe Author → LBFT (Either ErrLog Unit)
 verifyAuthorM author = do
   vs ← use (lSafetyRules ∙ srValidatorSigner)
-  maybeS vs (bail fakeErr) {-(ErrL (here' ["srValidatorSigner", "Nothing"]))-} $ λ validatorSigner →
-    maybeS
+  maybeS-RWST vs (bail fakeErr) {-(ErrL (here' ["srValidatorSigner", "Nothing"]))-} $ λ validatorSigner →
```

Comment:
It isn't necessary to use `maybeS-RWST` here, and even slightly preferable to use `maybeS`. This is generally true whenever the "scrutinee" (whatever we are doing case analysis on -- here `vs`) is exactly one of the variables that have been bound by a previous sequence.

Reply:
Thanks for this comment, the examples, and #108.  It would be good to capture the considerations here in comments more generally.  Is the following accurate/adequate?

> Within the `RWST` monad, `if` and `if-RWST` are semantically interchangeable, and similarly for `maybeS` and `maybe-RWST`.  The difference is in how proof obligations are generated, with the `*-RWST` variants generating new `weakestPre` obligations for each case.  In some cases, this is helpful for structuring proofs, while in other cases it is unnecessary and introduces more structure to the proof without adding any benefit.  A rule of thumb is that, if the ""scrutinee" (whatever we are doing case analysis on, i.e., the first argument) is the value provided via `RWST-bind` by a previous code block, then we already have a `weakestPre` proof obligation, so introducing additional ones via the `*-RWST` variants only creates more work and provides no additional benefit.

Reply:
Yes, that's an excellent description.

Reply:
Thanks for confirming @cwjnkins.  @haroldcarr, could you include the comment above before merging?  I'm thinking maybe in `RWST.Syntax`?

Reply:
I have added that description to Syntax.agda.